drivers/net/wireless/marvell/mwifiex/sdio.c

```diff
@@ -159,6 +159,59 @@ static const struct mwifiex_sdio_card_reg mwifiex_reg_sd8977 = {
 				 0x6a},
 };
 
+static const struct mwifiex_sdio_card_reg mwifiex_reg_sd8978 = {
+	.start_rd_port = 0,
+	.start_wr_port = 0,
+	.base_0_reg = 0xF8,
+	.base_1_reg = 0xF9,
+	.poll_reg = 0x5C,
+	.host_int_enable = UP_LD_HOST_INT_MASK | DN_LD_HOST_INT_MASK |
+		CMD_PORT_UPLD_INT_MASK | CMD_PORT_DNLD_INT_MASK,
+	.host_int_rsr_reg = 0x4,
+	.host_int_status_reg = 0x0C,
+	.host_int_mask_reg = 0x08,
+	.status_reg_0 = 0xE8,
+	.status_reg_1 = 0xE9,
+	.sdio_int_mask = 0xff,
+	.data_port_mask = 0xffffffff,
+	.io_port_0_reg = 0xE4,
+	.io_port_1_reg = 0xE5,
+	.io_port_2_reg = 0xE6,
+	.max_mp_regs = 196,
+	.rd_bitmap_l = 0x10,
+	.rd_bitmap_u = 0x11,
```

Comment:
Where all these values come from? Datasheet? Other mwifiex_reg structure?

Reply:
educated guess;
it's a copy of the datastructure for the "closest in name and function?" sd8997 - which seems to be also a sdio+uart wifi+bt combo chip 
and i didn't stumble over sth similar settings in the nxp driver... or at least nothing that seems different from how those two modules where handled there

Reply:
Question is: are these registers, values, masks correct? Can you verify that? Or at least can add a pointer/link to  your commit message pointing to the location which you used as base?

Reply:
added a note in the commit message - but as said: educated guess = copy of the SD8997 structure; which seems to work just find :P